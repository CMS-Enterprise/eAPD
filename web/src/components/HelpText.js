import React from 'react';
import PropTypes from 'prop-types';

import { t } from '../i18n';

const HelpText = ({ text: textResource, reminder: reminderResource }) => {
  const text = t(textResource, { defaultValue: false });
  const reminder = t(reminderResource, { defaultValue: false });

  return (
    (text || reminder) && (
      <div className="mb2">
        {text && <div>{text}</div>}
        {reminder && <div className="red">{reminder}</div>}
      </div>
    )
  );
};

HelpText.propTypes = {
  text: PropTypes.string.isRequired,
  reminder: PropTypes.string
};

HelpText.defaultProps = {
  reminder: null
};

export default HelpText;
