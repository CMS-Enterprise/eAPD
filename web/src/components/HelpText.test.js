import { shallow } from 'enzyme';
import React from 'react';

import HelpText from './HelpText';

describe('HelpText component', () => {
  test('renders correctly if title and reminder are emtpy', () => {
    const component = shallow(<HelpText text="something made up" />);
    expect(component).toMatchSnapshot();
  });

  test('renders correctly if title is not-empty but reminder is', () => {
    const component = shallow(<HelpText text="title" />);
    expect(component).toMatchSnapshot();
  });

  test('renders correctly if reminder is not-empty but title is', () => {
    const component = shallow(
      <HelpText text="something made up" reminder="title" />
    );
    expect(component).toMatchSnapshot();
  });

  test('renders correctly if title and reminder are not-empty', () => {
    const component = shallow(<HelpText text="title" reminder="title" />);
    expect(component).toMatchSnapshot();
  });
});
