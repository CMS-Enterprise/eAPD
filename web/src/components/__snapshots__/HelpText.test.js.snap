// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpText component renders correctly if reminder is not-empty but title is 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HelpText
    reminder="title"
    text="something made up"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="red"
        >
          [missing {{year}} value] HITECH APD
        </div>,
      ],
      "className": "mb2",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "[missing {{year}} value] HITECH APD",
          "className": "red",
        },
        "ref": null,
        "rendered": "[missing {{year}} value] HITECH APD",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="red"
          >
            [missing {{year}} value] HITECH APD
          </div>,
        ],
        "className": "mb2",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "[missing {{year}} value] HITECH APD",
            "className": "red",
          },
          "ref": null,
          "rendered": "[missing {{year}} value] HITECH APD",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HelpText component renders correctly if title and reminder are emtpy 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HelpText
    reminder={null}
    text="something made up"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HelpText component renders correctly if title and reminder are not-empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HelpText
    reminder="title"
    text="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          [missing {{year}} value] HITECH APD
        </div>,
        <div
          className="red"
        >
          [missing {{year}} value] HITECH APD
        </div>,
      ],
      "className": "mb2",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "[missing {{year}} value] HITECH APD",
        },
        "ref": null,
        "rendered": "[missing {{year}} value] HITECH APD",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "[missing {{year}} value] HITECH APD",
          "className": "red",
        },
        "ref": null,
        "rendered": "[missing {{year}} value] HITECH APD",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            [missing {{year}} value] HITECH APD
          </div>,
          <div
            className="red"
          >
            [missing {{year}} value] HITECH APD
          </div>,
        ],
        "className": "mb2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "[missing {{year}} value] HITECH APD",
          },
          "ref": null,
          "rendered": "[missing {{year}} value] HITECH APD",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "[missing {{year}} value] HITECH APD",
            "className": "red",
          },
          "ref": null,
          "rendered": "[missing {{year}} value] HITECH APD",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HelpText component renders correctly if title is not-empty but reminder is 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HelpText
    reminder={null}
    text="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          [missing {{year}} value] HITECH APD
        </div>,
        false,
      ],
      "className": "mb2",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "[missing {{year}} value] HITECH APD",
        },
        "ref": null,
        "rendered": "[missing {{year}} value] HITECH APD",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            [missing {{year}} value] HITECH APD
          </div>,
          false,
        ],
        "className": "mb2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "[missing {{year}} value] HITECH APD",
          },
          "ref": null,
          "rendered": "[missing {{year}} value] HITECH APD",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
